source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,582 @@
+
+(*) - optional attributes
+(**) - optional attributes that ***will not*** be implemented.
+
+### OVN ACL table
+
+An OVN ACL has the following attributes:
+
+| name      | type   | description |
+|-----------|--------|-------------|
+| priority  | int    | Range from 0 to 32,767. The smaller the value, the higher the priority                                                                                   |
+| direction | string | Either 'from-lport' or 'to-lport'.                                                                                                                       |
+| match     | string | The match criteria for the ACL. All the L3-L4 information (protocol, ether protocol, ports, ip prefix) is encoded here. Directly relates to OVS matches. |
+| action    | string | Either 'allow', 'allow-related', 'drop', or 'reject'. Currently, 'reject' behaves as 'drop'.                                                             |
+
+### Mapping the rules
+
+Mapping is a two way game: it is needed to translate *from* a networking
+API security group rule *to* an OVN ACL, ***and*** from an OVN ACL to a
+networking API security group rule.
+
+To achieve the latter, the information that is not **directly** encoded
+in the ACL - e.g. protocol, ethertype, ports (both min & max), security
+group id, remote ip prefix, and description - will be duplicated in the
+ACL external IDs.
+
+As the software matures, (most of) this information can eventually be
+extracted from the ACL match clause, but, in the first implementation,
+the plan is to duplicate it.
+
+The security group rule **white list** behavior will be implemented
+through the priority concept; throughout the code two different
+priorities will be used: a lesser priority for **all** ACLs having a
+deny *action* and an higher priority for **all** ACLs having *allow* or
+*allow-related* actions.
+
+## Activating the feature
+
+The feature will be always active, and it is a matter of identifying the
+ports that will be subject to the security group filters.
+
+As previously mentioned, the **port-security** attribute will be used to
+know if a port will be added to the **deny-all** port group, having all
+IP traffic dropped as a consequence.
+
+The **port-security** attribute is set at port creation time, and its value will
+be inherited from the corresponding *network* entity when not defined at port
+level. Afterwards, it can be updated through the API - at port *or* network
+levels.
+
+The default behavior for the network's **port-security** attribute would be
+defined in the configuration file, through the ***port-security-enabled***,
+property, located in a new section, called **NETWORK**. This value will be set
+during *engine-setup* - refer to [this section](#engine-setup) - and the
+default choice will be 'yes', which will result in all traffic being dropped to
+*and* from the VMs.
+
```

Comment:
If security groups are enabled during engine-setup, shouldn't this result in something like the [default config of OpenStack](https://imgur.com/a/2UWPgau)

Reply:
I agree with that observation - but propose a twist to it.

Looking at the rules you provide, it looks like **all egress traffic is allowed**,whereas **all ingress** from instances ***belonging*** to the **'default' security group is allowed**.

In this feature page, the implementation of the 'remote_security_group' parameter is discarded, since it would bloat the feature a lot - and this is actually the first motive for including it I've found. Never-the-less, I agree that's a good motive to include it.

I suggest we mimic the egress behavior - allow outgoing traffic - **but** drop all incoming traffic by default.